### PR TITLE
Improve actor and http api server metrics

### PR DESCRIPTION
**Pull request by Andyz26**
### Context

* New metrics in http server/routes.
* Attach metered mailbox to resource cluster actor.

### Checklist

- [x] `./gradlew build` compiles code correctly
- [x] Added new tests where applicable
- [x] `./gradlew test` passes all tests
- [x] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/MasterMain.java [hunk 1]
```diff
@@ -365,6 +367,7 @@ public static void main(String[] args) {
         }
 
         try {
+            SpectatorRegistryFactory.setRegistry(new DefaultRegistry());
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/MasterMain.java:370]: why is this needed? Even if it's required, wouldn't it make sense to do this in nfmantis similar to worker-titus.

File: mantis-common/src/main/java/io/mantisrx/common/metrics/LoggingMetricsPublisher.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This implementation will print the selected metric groups in log.
+ * To use this publisher override config "mantis.metricsPublisher.class" in the master-*.properties file.
+ */
+@Slf4j
+public class LoggingMetricsPublisher extends MetricsPublisher {
```
Comment (sundargates) [inline mantis-common/src/main/java/io/mantisrx/common/metrics/LoggingMetricsPublisher.java:34]: just curious - why is this required? we are anyways publishing these metrics today via spectator - what's the need to log it.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/MasterApiAkkaService.java [hunk 3]
```diff
@@ -209,18 +208,22 @@ private void startAPIServer() {
...
+                }))
+                .run(materializer)
+                .exceptionally(failure -> {
+                    System.err.println("API service exited, committing suicide !" + failure.getMessage());
+                    logger.info("Master API service exited in error, committing suicide !");
+                    system.terminate();
+                    System.exit(2);
+                    return null;
+                });
         logger.info("Starting Mantis Master API on port {}", port);
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/MasterApiAkkaService.java:227]: What is this for?

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/MasterApiMetrics.java [hunk 4]
```diff
@@ -24,13 +24,15 @@ public class MasterApiMetrics {
     private final Counter resp2xx;
     private final Counter resp4xx;
     private final Counter resp5xx;
+    private final Counter incomingRequestCount;
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/MasterApiMetrics.java:27]: It looks like this is across all requests. Is there a way to emit this with the route path as a tag?

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java [hunk 5]
```diff
@@ -117,7 +117,8 @@ class ResourceClusterActor extends AbstractActorWithTimers {
     private final String jobClustersWithArtifactCachingEnabled;
 
     static Props props(final ClusterID clusterID, final Duration heartbeatTimeout, Duration assignmentTimeout, Duration disabledTaskExecutorsCheckInterval, Clock clock, RpcService rpcService, MantisJobStore mantisJobStore, JobMessageRouter jobMessageRouter, int maxJobArtifactsToCache, String jobClustersWithArtifactCachingEnabled) {
-        return Props.create(ResourceClusterActor.class, clusterID, heartbeatTimeout, assignmentTimeout, disabledTaskExecutorsCheckInterval, clock, rpcService, mantisJobStore, jobMessageRouter, maxJobArtifactsToCache, jobClustersWithArtifactCachingEnabled);
+        return Props.create(ResourceClusterActor.class, clusterID, heartbeatTimeout, assignmentTimeout, disabledTaskExecutorsCheckInterval, clock, rpcService, mantisJobStore, jobMessageRouter, maxJobArtifactsToCache, jobClustersWithArtifactCachingEnabled)
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java:120]: This can be configured directly in `application.conf`.

File: mantis-common/src/main/java/io/mantisrx/common/metrics/LoggingMetricsPublisher.java (hunk 2 above)
Comment (sundargates) [reply mantis-common/src/main/java/io/mantisrx/common/metrics/LoggingMetricsPublisher.java:34]: If this is only expected to be used for testing purposes, I recommend that you move this class to the test package and make it package private.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/MasterMain.java (hunk 1 above)
Comment (sundargates) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/MasterMain.java:370]: If this is for test containers, I would recommend moving this there since this module is not expected to be used as the leaf node.

Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/MasterMain.java:370]: in default masterMain-> main the spectator's default composite registry is empty thus no metrics is actually logged. This is needed for any app using this entry to be able to actually log metrics there.

File: mantis-common/src/main/java/io/mantisrx/common/metrics/LoggingMetricsPublisher.java (hunk 2 above)
Comment (Andyz26) [reply mantis-common/src/main/java/io/mantisrx/common/metrics/LoggingMetricsPublisher.java:34]: It is a debug mode but to use this it needs to be configured via the core config (this interface is one of these created in config instead of injection).
Also in OSS domain, I think it's okay to have a logging-based mode option.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/MasterApiAkkaService.java (hunk 3 above)
Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/MasterApiAkkaService.java:227]: Remove deprecated akka api + log incrementIncomingRequestCount.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/MasterApiMetrics.java (hunk 4 above)
Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/MasterApiMetrics.java:27]: This is used at the initial connection point so the path could be anything (cardinality not controlled by us). we have some route-level metrics already at each route level but it's up to the route to decide whether to log so I want something in general for all connections (plus track if something keeps requesting for invalid paths).

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java (hunk 5 above)
Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java:120]: let's keep it consistent with other usages e.g. jobManagerClusterActor?
